internal/xds/translator/translator.go

```diff
@@ -500,6 +572,21 @@ func (t *Translator) addRouteToRouteConfig(
 		// If no extension exists (or it doesn't subscribe to this hook) then this is a quick no-op.
 		if err = processExtensionPostVHostHook(vHost, t.ExtensionManager); err != nil {
 			errs = errors.Join(errs, err)
+			if t.ExtensionManager != nil && !(*t.ExtensionManager).FailOpen() {
+				vHost.Routes = []*routev3.Route{
```

Comment:
would be good to have some comments in here explaining what this setting means, and why its added here, similar comment would be great for other snippets where xds is being modified to return a 500

Reply:
Done